layers/best_practices_error_enums.h

```
@@ -36,6 +36,10 @@ static const char DECORATE_UNUSED *kVUID_BestPractices_CreateInstance_Deprecated
     "UNASSIGNED-BestPractices-vkCreateInstance-deprecated-extension";
 static const char DECORATE_UNUSED *kVUID_BestPractices_CreateDevice_DeprecatedExtension =
     "UNASSIGNED-BestPractices-vkCreateDevice-deprecated-extension";
+static const char DECORATE_UNUSED *kVUID_BestPractices_CreateInstance_SpecialUseExtension =
```

Since we plan on telling people who are actually suppose to use these extensions to "just ignore the error message" I would purpose we have a single `kVUID_BestPractices_SpecialUseExtension ` or something and then pass a `bool instance` into `ValidateSpecialUseExtensions` so that the error message says `instance` or `device`.

Mainly it will be easier to have a single warning message to ignore for those who are suppose too

---

I had it this way originally, but deferred because of the structure of actual VUIDs.  If they ever do get formalized, would they not still have the API name/structure name as part of the VUID?  If it doesn't matter I'm happy to change it back!

---

yes, I guess in the future would be more formal and the name would change (for better or worst)

I guess maybe worth adding a note that this is a special case as its an VUID that we expect a good number of people to ignore and therefore made it a single VUID string

@Tobski thoughts cause always enjoy more opinions on these trivial things that once decided, are never worth the time to change again

---

Remember, this is best practices which is opt-in and wouldn't normally be running with normal validation. Might affect your opinion of the noise level.

---

Sounds like there's probably an opportunity to fix this if/when the VUID names get formalised, so I don't think it matters much here?

---

🚢 it!